src/LiveComponent/assets/src/Component/index.ts

```diff
@@ -300,15 +300,60 @@ export default class Component {
 
         this.backendRequest.promise.then(async (response) => {
             const backendResponse = new BackendResponse(response);
-            const html = await backendResponse.getBody();
 
             // clear sent files inputs
             for (const input of Object.values(this.pendingFiles)) {
                 input.value = '';
             }
 
-            // if the response does not contain a component, render as an error
             const headers = backendResponse.response.headers;
+            if (headers.get('X-Live-Download')) {
+                if (
+                    !(
+                        headers.get('Content-Disposition')?.includes('attachment') ||
+                        headers.get('Content-Disposition')?.includes('inline')
+                    ) ||
+                    !headers.get('Content-Disposition')?.includes('filename=')
+                ) {
+                    throw new Error('Invalid LiveDownload response');
+                }
+
+                const fileSize = Number.parseInt(headers.get('Content-Length') || '0');
+                if (fileSize > 10000000) {
+                    throw new Error('File is too large to download (10MB limit)');
+                }
```

Comment:
More correct and easier to understand:
```suggestion
                if (fileSize > 10 * 1024 * 1024) {
                    throw new Error('File is too large to download (10MB limit)');
                }
```

Also, why do you want to limit the file size? It can be a nice feature, but shouldn't it be configurable by the developer?

Reply:
This is the part we need to discuss. Don't forget the user has NOT accepted this download in any way for now.

Also, you cannot store 1GB in the DOM before saving the file.

We have options after: 
- streaming toward the FileStorage api (where user would then decide where and how to save the file)
- allowing this only on button with attribute "download" (my preference for now)

Reply:
No limit